src/commands/workspace/__tests__/add.test.js

```diff
@@ -14,14 +14,34 @@ jest.mock('../../../utils/logger');
 const unsafeProcessses: any & typeof processes = processes;
 const unsafeYarn: any & typeof yarn = yarn;
 
-async function depIsInstalled(workspaceDir: string, depName: string) {
+// Helper method to check if a dependency is installed, both in the package.json and on the fs
+async function depIsInstalled(
+  workspaceDir: string,
+  depName: string,
+  version?: string
+) {
   const pkg = await Package.init(path.join(workspaceDir, 'package.json'));
   const dirExists = await pathExists(
     path.join(workspaceDir, 'node_modules', depName)
   );
   const depInPkgJson = pkg.getDependencyType(depName) !== null;
+  const correctVersion =
+    !version || pkg.getDependencyVersionRange(depName) === version;
 
-  return dirExists && depInPkgJson;
+  return dirExists && depInPkgJson && correctVersion;
+}
+
+// a mock yarn add function to update the pakcages's config and also node_modules dir
+async function fakeYarnAdd(pkg: Package, dependencies, type = 'dependencies') {
```

Comment:
I just copied the two helped functions from the other tests as they were a bit nicer